src/frontend/xgboost/xgboost_json.h

```diff
@@ -0,0 +1,376 @@
+ public:
+  using OutputHandler<std::vector<ElemType>>::OutputHandler;
+  bool Bool(ElemType b) {
+    this->m_output.push_back(b);
+    return true;
+  }
+  template <typename ArgType, typename IntType = ElemType>
+  typename std::enable_if<std::is_integral<IntType>::value, bool>::type
+  store_int(ArgType i) {
+    this->m_output.push_back(static_cast<ElemType>(i));
+    return true;
+  }
+
+  template <typename ArgType, typename IntType = ElemType>
+  typename std::enable_if<!std::is_integral<IntType>::value, bool>::type
+  store_int(ArgType i) {
+    return false;
+  }
+
+  bool Int(int i) { return store_int<int>(i); }
+  bool Uint(unsigned u) { return store_int<unsigned>(u); }
+  bool Int64(int64_t i) { return store_int<int64_t>(i); }
+  bool Uint64(uint64_t u) { return store_int<uint64_t>(u); }
+  bool Double(ElemType d) {
+    this->m_output.push_back(d);
+    return true;
+  }
+
+  template <typename StringType = ElemType>
+  typename std::enable_if<std::is_same<StringType, std::string>::value,
+                          bool>::type
+  store_string(const char *str, std::size_t length, bool copy) {
+    this->m_output.push_back(ElemType{str, length});
+    return true;
+  }
+  template <typename StringType = ElemType>
+  typename std::enable_if<!std::is_same<StringType, std::string>::value,
+                          bool>::type
+  store_string(const char *str, std::size_t length, bool copy) {
+    return false;
+  }
+
+  bool String(const char *str, std::size_t length, bool copy) {
+    return store_string(str, length, copy);
+  }
+
+  bool StartObject(std::true_type) {
+    this->m_output.emplace_back();
+    return this->template push_handler<HandlerType, ElemType>(
+        this->m_output.back());
+  }
+
+  bool StartObject(std::false_type) { return false; }
+
+  bool StartObject() {
+    return StartObject(
+        std::integral_constant<bool, std::is_base_of<OutputHandler<ElemType>,
+                                                     HandlerType>::value>{});
+  }
+};
+
+/*! \brief handler for TreeParam objects from XGBoost schema*/
+class TreeParamHandler : public OutputHandler<int> {
+ public:
+  using OutputHandler<int>::OutputHandler;
+
+  bool String(const char *str, std::size_t length, bool copy);
+};
+
+/*! \brief handler for RegTree objects from XGBoost schema*/
+class RegTreeHandler : public OutputHandler<treelite::Tree> {
+ public:
+  using OutputHandler<treelite::Tree>::OutputHandler;
+  bool StartArray();
+
+  bool StartObject();
+
+  bool Uint(unsigned u);
+
+  bool EndObject(std::size_t memberCount);
+
+ private:
+  std::vector<double> loss_changes;
+  std::vector<double> sum_hessian;
+  std::vector<double> base_weights;
+  std::vector<int> leaf_child_counts;
+  std::vector<int> left_children;
+  std::vector<int> right_children;
+  std::vector<int> parents;
+  std::vector<int> split_indices;
+  std::vector<double> split_conditions;
+  std::vector<bool> default_left;
+  int num_nodes;
```

Comment:
```suggestion
  std::vector<double> m_loss_changes;
  std::vector<double> m_sum_hessian;
  std::vector<double> m_base_weights;
  std::vector<int> m_leaf_child_counts;
  std::vector<int> m_left_children;
  std::vector<int> m_right_children;
  std::vector<int> m_parents;
  std::vector<int> m_split_indices;
  std::vector<double> m_split_conditions;
  std::vector<bool> m_default_left;
  int m_num_nodes;
```
to be consistent with `m_output`